src/partition/repart.c

```diff
@@ -6033,11 +6034,7 @@ static int resize_pt(int fd) {
          * possession of the enlarged backing file. For this it suffices to open the device with libfdisk and
          * immediately write it again, with no changes. */
 
-        c = fdisk_new_context();
-        if (!c)
-                return log_oom();
-
-        r = fdisk_assign_device(c, FORMAT_PROC_FD_PATH(fd), 0);
+        r = fdisk_new_context_fd(fd, fd, /* read_only= */ false);
```

Comment:
The arguments are borked, and this triggers assertion.

Reply:
ouch!